libcxx/include/optional

```diff
@@ -983,72 +984,133 @@ public:
 template <class _Tp>
 optional(_Tp) -> optional<_Tp>;
 
-// Comparisons between optionals
+// [optional.relops] Relational operators
+
+#    if _LIBCPP_STD_VER >= 26
+template < class _Tp, class _Up>
+#    else
 template <
     class _Tp,
     class _Up,
     enable_if_t<is_convertible_v<decltype(std::declval<const _Tp&>() == std::declval<const _Up&>()), bool>, int> = 0>
```

Comment:
This is curious. We've been constraining these comparison operators since initially implementing `optional` (a9e659619f592c49b352af16253ec0b79c207956, https://reviews.llvm.org/D22741), although they were not constrained in the standard until C++26.

I guess we should either impose equivalent SFINAE constraints in all modes or remove the constraints before C++26.

Reply:
Removing these constraints feel like the less painful option but I'll need a maintainer consensus on how to proceed.

Reply:
It seems better to split changes for `optional` comparison operators to another PR, so that we can proceed more quickly.

Reply:
Removed:
- [optional.relops](https://wg21.link/optional.relops)
- [optional.comp.with.t](https://wg21.link/optional.comp.with.t)